src/material/datepicker/multi-year-view.ts

```diff
@@ -39,6 +39,43 @@ export const yearsPerPage = 24;
 
 export const yearsPerRow = 4;
 
+/** Gets remainder that is non-negative, even if first number is negative */
+function euclideanModulo (a: number, b: number): number {
+  return (a % b + b) % b;
+}
+
+/**
+ * We pick a "starting" year such that either the maximum year would be at the end
+ * or the minimum year would be at the beginning of a page.
+ */
+function getStartingYear(minDate: any, maxDate: any): number {
+  let startingYear = 0;
+  if (maxDate) {
+    const maxYear = maxDate.getFullYear();
+    startingYear = maxYear - yearsPerPage + 1;
+  } else if (minDate) {
+    startingYear = minDate.getFullYear();
+  }
+  return startingYear;
+}
+
+/**
+ * When the multi-year view is first opened, the active year will be in view.
+ * So we compute how many years are between the active year and the *slot* where our
+ * "startingYear" will render when paged into view.
+ */
+export function getActiveOffset(activeDate: any, minDate: any, maxDate: any): number {
```

Comment:
If there are new exported helper functions like this, then they need to be omitted from `datepicker/public-api.ts` in order to avoid introducing them as public APIs.